prover/src/opcodes/groestl.rs

```diff
@@ -0,0 +1,815 @@
+use std::{array::from_fn, cell::RefMut};
+
+use binius_field::packed::set_packed_slice;
+use binius_field::AESTowerField8b;
+use binius_field::Field;
+use binius_hash::groestl::GroestlShortImpl;
+use binius_hash::groestl::GroestlShortInternal;
+use binius_m3::builder::Expr;
+use binius_m3::{
+    builder::{
+        upcast_col, Col, ConstraintSystem, TableFiller, TableId, TableWitnessSegment, B32, B64, B8,
+    },
+    gadgets::hash::groestl::Permutation,
+};
+use petravm_asm::{Groestl256CompressEvent, Groestl256OutputEvent, Opcode};
+
+use crate::gadgets::transpose::TransposeColumns;
+use crate::utils::u64_to_bytes;
+use crate::utils::u64_to_u32;
+use crate::{
+    channels::Channels,
+    gadgets::state::{NextPc, StateColumns, StateColumnsOptions, StateGadget},
+    table::Table,
+    types::ProverPackedField,
+};
+
+const GROESTL_COMPRESS_OPCODE: u16 = Opcode::Groestl256Compress as u16;
+const GROESTL_OUTPUT_OPCODE: u16 = Opcode::Groestl256Output as u16;
+
+/// GROEST256_COMPRESS table.
+///
+/// This table handles the GROEST256_COMPRESS instruction, which performs the
+/// compression function described in the Groestl specs.
```

Comment:
Maybe mention here the transformation applied to the data stored in the VROM, which is required in order to use the permutation gadgets